arangod/RestHandler/RestStatusHandler.cpp

```diff
@@ -274,3 +285,26 @@ RestStatus RestStatusHandler::executeOverview() {
   generateResult(rest::ResponseCode::OK, result.slice());
   return RestStatus::DONE;
 }
+
+RestStatus RestStatusHandler::executeMemoryProfile() {
+#if defined(USE_MEMORY_PROFILE)
+  std::string const outDirectory = TRI_GetTempPath();
+  std::string const fileName = FileUtils::buildFilename(outDirectory, "profile.out");
+  char const* f = fileName.c_str();
+  try {
+    mallctl("prof.dump", NULL, NULL, &f, sizeof(const char *));
+    std::string const content = FileUtils::slurp(fileName);
+    TRI_UnlinkFile(f);
+
+    resetResponse(rest::ResponseCode::OK);
+
+    _response->setContentType(rest::ContentType::TEXT);
+    _response->addRawPayload(velocypack::StringRef(content));
+  } catch (...) {
+    TRI_UnlinkFile(f);
+    throw;
+  }
+
+#endif
+  return RestStatus::DONE;
```

Comment:
In case we get here, but `USE_MEMORY_PROFILE` was not defined, the handler will not return any response (not even an error). This is currently prevented by an earlier check in https://github.com/arangodb/arangodb/pull/12950/files#diff-61ec4959a4f0db4cf223f986abc0033d454368ff6825efb23c30857402f58b64R78-R81, but it requires looking at two different places. 
I suggest we add an explicit `not implemented` or `not found` response here for the case in which `USE_MEMORY_PROFILE` is not defined. Then we can also remove the other check and have all the handling in one place.

Reply:
Done